src/build/optimize-streams.ts

```diff
@@ -75,26 +82,46 @@ export class GenericOptimizeStream extends Transform {
 }
 
 /**
- * JSOptimizeStream optimizes JS files that pass through it (via uglify).
+ * JSBabelTransform uses babel to transpile Javascript, most often rewriting
+ * newer ECMAScript features to only use language features available in major
+ * browsers. If no options are given to the constructor, JSBabelTransform will
+ * use
+ * a babel's default "ES6 -> ES5" preset.
  */
-export class JSOptimizeStream extends GenericOptimizeStream {
-  constructor(options: UglifyOptions) {
-    // uglify is special, in that it returns an object with a code property
-    // instead of just a code string. We create a compliant optimizer here
-    // that returns a string instead.
-    const uglifyOptimizer = (contents: string, options: UglifyOptions) => {
-      return uglify(contents, options).code;
+class JSBabelTransform extends GenericOptimizeTransform {
+  constructor(config: BabelTransformOptions) {
+    let transform = (contents: string, options: BabelTransformOptions) => {
+      return babelTransform(contents, options).code!;
```

Comment:
no, just errors. Discussed in person, but we may need to turn these into true stream errors if code doesn't compile. That comes with its own set of problems (should we block if dependencies don't compile?). 

Will monitor and address if necessary.